libr/anal/dwarf_process.c

```diff
@@ -1188,9 +1188,11 @@ static st32 parse_function_args_and_vars(Context *ctx, ut64 idx, RStrBuf *args,
 					}
 					var->type = strdup (r_strbuf_get (&type));
 					r_list_append (variables, var);
+				} else {
+					free (var->location);
```

Comment:
i'd prefer to see a local function `variable_free (Variable *)` that correctly frees the whole structure and that is used here, later and also in `parse_function`.

Reply:
True, done